pkg/deploy/controller/deploymentconfig/controller.go

```diff
@@ -93,7 +105,7 @@ func (c *DeploymentConfigController) Handle(config *deployapi.DeploymentConfig)
 			if !deployutil.IsDeploymentCancelled(&deployment) {
 				deployment.Annotations[deployapi.DeploymentCancelledAnnotation] = deployapi.DeploymentCancelledAnnotationValue
```

Comment:
EEEK!  You've got a shallow copy, so this map (pointer) is the same.  I think you just overwrote the cache here.  If you make a mutation, make a deep copy.  `kapi.Scheme.Copy` I think.